DOCUMENTATION.md

```diff
@@ -722,23 +693,74 @@ If the request to the client is successful, the client returns a `dict`.
-|`403`|`[{`<br>`"error": "AuthError",`<br>`"message": "Invalid token: API key not found"`<br>`}]`|Use the correct API key. Refer to [API keys](#api-keys) for more information|
+|`400`|`[{`<br>`"error": "ValidationError",`<br>`"message": "status ‘elephant’ is not one of [cancelled, created, sending, sent, delivered, pending, failed, technical-failure, temporary-failure, permanent-failure, pending-virus-check, validation-failed, virus-scan-failed, returned-letter, accepted, received]"`<br>`}]`|Change the [status argument](#status-optional)|
+|`400`|`[{`<br>`"error": "ValidationError",`<br>`"message": "‘Apple’ is not one of [sms, email, letter]"`<br>`}]`|Change the [template_type argument](#template-type-optional)|
 
+## Email status descriptions
+
+|Status|Description|
+|:---|:---|
+|`created`|GOV.UK Notify has placed the message in a queue, ready to be sent to the provider. It should only remain in this state for a few seconds.|
+|`sending`|GOV.UK Notify has sent the message to the provider. The provider will try to deliver the message to the recipient for up to 72 hours. GOV.UK Notify is waiting for delivery information.|
+|`delivered`|The message was successfully delivered.|
+|`failed`| This covers all 3 failure statuses: `permanent-failure`, `temporary-failure` and `technical-failure`.|
+|`permanent-failure`|The provider could not deliver the message because the email address was wrong. You should remove these email addresses from your database.|
+|`temporary-failure`|The provider could not deliver the message. This can happen when the recipient’s inbox is full. You can try to send the message again.|
+|`technical-failure`|Your message was not sent because there was a problem between Notify and the provider.<br>You’ll have to try sending your messages again.|
+
+## Text message status descriptions
+
+|Status|Description|
+|:---|:---|
+|`created`|GOV.UK Notify has placed the message in a queue, ready to be sent to the provider. It should only remain in this state for a few seconds.|
+|`sending`|GOV.UK Notify has sent the message to the provider. The provider will try to deliver the message to the recipient for up to 72 hours. GOV.UK Notify is waiting for delivery information.|
+|`pending`|GOV.UK Notify is waiting for more delivery information.<br>GOV.UK Notify received a callback from the provider but the recipient’s device has not yet responded. Another callback from the provider determines the final status of the text message.|
+|`sent`|The message was sent to an international number. The mobile networks in some countries do not provide any more delivery information. The GOV.UK Notify website displays this status as 'Sent internationally'.|
+|`delivered`|The message was successfully delivered.|
+|`failed`| This covers all 3 failure statuses: `permanent-failure`, `temporary-failure` and `technical-failure`.|
```

Comment:
From what I can see, `failed` is a deprecated status so I don't think we will return this anymore from the API? And therefore it wouldn't make sense to have a row for `failed` in any of the status tables. Is that right @servingUpAces 

https://github.com/alphagov/notifications-api/blob/32f5f454dec31b540e39e27f9fd845298f15ae2b/app/models.py#L1466

Reply:
After talking to Leo and David, we removed the `failed` status from the status description tables because:

* this is not a status that will be returned by the API
* the purpose of the tables is to describe statuses that can be returned by the API

We moved `failed` to the arguments for How to get the status of multiple messages, for users who want to know how to return all failure statuses.